src/core/control/jobs/RenderJob.cpp

```diff
@@ -79,12 +80,32 @@ void RenderJob::run() {
 
         std::lock_guard lock(this->view->drawingMutex);
         std::swap(this->view->crBuffer, newBuffer);
+        repaintPage();
     } else {
-        for (Rectangle<double> const& rect: rerenderRects) { rerenderRectangle(rect); }
+        for (Rectangle<double> const& rect: rerenderRects) {
+            rerenderRectangle(rect);
+            repaintPageArea(rect.x, rect.y, rect.x + rect.width, rect.y + rect.height);
+        }
     }
+}
+
+void RenderJob::repaintPage() {
+    int x1 = view->getX();
+    int y1 = view->getY();
+    int x2 = x1 + view->getDisplayWidth();
+    int y2 = y1 + view->getDisplayHeight();
+    repaintWidgetArea(view->xournal->getWidget(), x1, y1, x2, y2);
```

Comment:
I think can be simplified to `repaintPageArea(0, 0, view->getWidth(), view->getHeight())`